### ISSUE TITLE
Unable to add enchantment off of a book to a sword 

### ISSUE THREAD

**Issue by GameRundown** [Fourmisain/AxesAreWeapons#20]
I installed the mod and noticed that I couldn't pull knockback off a book to my sword. You'd normally be able to get both enchantments off the book if possible or the ones that can be used for your weapon. In this case I should be able to get only the knockback off the book for my sword.

![2024-09-23_11 15 37](https://github.com/user-attachments/assets/288e5681-e67c-4e6e-bb3e-ce2af438c501)
![2024-09-23_11 15 42](https://github.com/user-attachments/assets/828da1ce-10c9-46a8-b0e2-1bdf6813e550)

**Comment by Fourmisain** [comment]
Hm... at least on 1.21, I can't seem to reproduce the issue:

![1](https://github.com/user-attachments/assets/b679b112-f0ee-47b1-88dc-12080b19c435)
![2](https://github.com/user-attachments/assets/ec144b84-ef8d-47d0-b048-4447c17b04d4)

What version are you playing on? Are you using any other mods?

**Comment by GameRundown** [comment]
I'm playing on 1.21.1. I'm on Fabric. I am using some other mods. This was on my server. Could it be only related to the server? I'll see if the issue happens on single player too. I added this mod and the problem appeared and I removed it and it went away. I'll also try using the bare minimum amount of mods for Fabric and see if there is a conflict with any of my mods. I'm not using any mods similar to Axes are Weapons. Or any enchantment mods

**Comment by GameRundown** [comment]
Can confirm the issue only happens on my server. I just tried it in single player and it worked fine.

**Comment by Fourmisain** [comment]
I just tried running Axes Are Weapons on a 1.21 server and connecting with a vanilla client, but couldn't reproduce the issue.
Tried with `enableKnockback` on and off too.

It's likely a mod conflict of some kind - something related to swords or tags probably.
(Axes Are Weapons does change the Knockback enchantment to use a different tag, which contains the vanilla `enchantable/sword` tag, but if that somehow fails, it would explain why swords cannot be enchanted with Knockback anymore.)

A mod list would be good so I can give it a look.
It's probably enough if you pull the list out of the log files.

**Comment by GameRundown** [comment]
Okay, so I tried using the bare minimum amount of mods. Axes Are Weapons, Fabric api, Indium, Iris, Sodium, and Cloth Config. I still had the same issue. I think I am narrowing it down though. The issue is only on my server which I have through Apex Hosting. It seems like knockback is the only enchantment that is broken on the server. I was able to use sharpness if I wanted to.

![2024-09-23_18 24 18](https://github.com/user-attachments/assets/af1494e5-93f1-4f62-b1ff-452d56cc900f)
![2024-09-23_18 24 09](https://github.com/user-attachments/assets/66f3d862-2fdc-415c-bee5-5281be5f40b5)

**Comment by GameRundown** [comment]
Oh sorry you are right those are client side mods. The only mod on the server should be the Axes Are Weapons mod and Cloth Config. this might be the issue. Our world was on Paper and I tried switching it to Fabric. It might have messed up the world. When I showed the list it had Bukkit as a data pack for some reason.

![2024-09-24_08 06 50](https://github.com/user-attachments/assets/b5f37423-9a0d-47d5-b65f-b8bf4646d11d)

**Comment by Fourmisain** [comment]
> Our world was on Paper and I tried switching it to Fabric. It might have messed up the world.

Ohh, that would explain it! It looks like Fabric isn't installed properly, as you should be seeing Fabric datapacks.
I'm not sure how Apex Hosting works, but it might have tried installing Fabric on top of Paper, when it needs to be installed on top of Vanilla or something along those lines.

You should be able to fix it by first downloading a backup of your world and all other server files you might need (like `server.properties`, `whitelist.json`, `ops.json`, the `config`, and `mods` folders).
Inside your world folder should be a `datapacks` folder containing the `bukkit` datapack, which you should probably delete (or just backup somewhere else). Not entirely sure what it does, but you definitely don't want to use it on Fabric.

After your backup's done, set up a new server instance with Fabric and upload all your server files onto that.

Would still be interested in seeing the server log, curious what it would say.

**Comment by GameRundown** [comment]
Sorry for the wait. I was working out of town. Here is the latest server log. 

[latest.log](https://github.com/user-attachments/files/17171485/latest.log)

**Comment by GameRundown** [comment]
I just made a whole new world with fabric. The only thing listed is vanilla. Why is that? Here is the server log for the new world as well.

![2024-09-27_19 25 33](https://github.com/user-attachments/assets/a2b60067-a136-4955-8fd9-a924971aaa27)
[latest (3).log](https://github.com/user-attachments/files/17171671/latest.3.log)

**Comment by GameRundown** [comment]
I finally figured it out! It was because I didn't have Fabric API installed on my server. So, the Axes Are Weapons mod wasn't completely working. That is also the reason why my /datapack list looked different too. I definitely should have just had API installed but I would list it as a dependency for your mod along with cloth config.

**Comment by Fourmisain** [comment]
Wow, you're absolutely right! I had no idea that's a thing!

It's specifically the [`fabric-resource-loader` module](https://github.com/FabricMC/fabric/tree/master/fabric-resource-loader-v0) of Fabric API that's responsible to load mod datapacks.
Kinda weird that Fabric _Loader_ needs the API to _load_ the datapack part, but it also makes sense because Fabric Loader is supposed to be minimal.

When I tried testing your solution and disabled Fabric API I was confused because it still worked - turns out Mod Menu contains that specific module as well!

This issue first appeared in version 1.8.0 (Minecraft 1.20.5+), older versions are fine.
I just added Fabric API as a depencency on Modrinth and CurseForge and added it to the description as well.
Next version will have a hard-dependency on it.

Sorry you had to do all the hard work to figure it out!
